### DISCUSSION TITLE
DNS rebinding

### DISCUSSION THREAD

**Discussion by noraj** [projectdiscovery/interactsh#317] [Ideas]
As Interactsh is about OOB interaction supporting DNS it would be nice to have a DNS rebinding by the same occasion.

This mean having a way to get resolvable resource records with any given contents for assisting in detection and exploitation of SSRF-related vulnerabilities.

I have fetched a [list of DNS rebinding](https://gitlab.com/rawsec/rawsec-cybersecurity-list/-/issues/284) tools but most are hard to setup, featureless and unmaintained. The most interesting project to understand all major features is [1u.ms](http://1u.ms/) ([source](https://github.com/neex/1u.ms)), their website describe well all possibilities.

Right now Interactsh is similar to most HTTP request collector and inspector tools but lacks of **DNS rebinding**.

![image](https://user-images.githubusercontent.com/16578570/121225988-59a59d00-c88a-11eb-803d-357b3d371c21.png)

What I said for DNS could also be meaningful for HTTP, being able choose the body or headers served by the server.

**Comment by Mzack9999** [comment]
Implemented in https://github.com/projectdiscovery/interactsh/issues/29

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
